tools/linters/cpplint_wrapper.py

```diff
@@ -0,0 +1,127 @@
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+from roslint import cpplint
+from roslint.cpplint import Match, IsBlankLine, main
+from functools import partial
+
+import os.path
+import re
+
+
+def patch(original_module):
+    """ Decorator to easily allow wrapping/overriding of the Check* functions in cpplint. Should
+        decorate a function which matches the signature of the function it replaces expect with
+        the addition of a fn parameter, which is a pass-through of the replaced function, in case
+        the replacement would like call through to the original functionality. """
+    def wrap(override_fn):
+        original_fn = getattr(original_module, override_fn.__name__)
+        setattr(original_module, override_fn.__name__, partial(override_fn, original_fn))
+
+        # Don't actually modify the function being decorated.
+        return override_fn
+    return wrap
+
+
+def makeErrorFn(original_fn, suppress_categories, suppress_message_matches):
+    """ Create a return a wrapped version of the error-report function which suppresses specific
+        error categories. """
+    def newError(filename, linenum, category, confidence, message):
+        if category in suppress_categories:
+            return
+        if True in [bool(Match(r, message)) for r in suppress_message_matches]:
+            return
+        original_fn(filename, linenum, category, confidence, message)
+    return newError
+
+
+@patch(cpplint)
+def GetHeaderGuardCPPVariable(fn, filename):
+    # Overload the Header Guard Check to only use the filename, not the path
+    head, tail = os.path.split(filename)
+    return re.sub(r'[-./\s]', '_', tail.upper())
+
+
+@patch(cpplint)
+def CheckBraces(fn, filename, clean_lines, linenum, error):
+    """ Complete replacement for cpplint.CheckBraces, since the brace rules for ROS C++ Style
+        are completely different from the Google style guide ones. """
+    line = clean_lines.elided[linenum]
+    if Match(r'^(.*){(.*)}.?$', line):
+        # Special case when both braces are on the same line together, as is the
+        # case for one-line getters and setters, for example, or rows of a multi-
+        # dimenstional array initializer.
+        pass
+    else:
+        # Line does not contain both an opening and closing brace.
+        m = Match(r'^(.*){(.*)$', line)
+        if m and not (IsBlankLine(m.group(1))):
+            # Line contains a starting brace and is not empty, uh oh.
+            if "=" in line:
+                # Opening brace is permissable in case of an initializer.
+                pass
+            else:
+                error(filename, linenum, 'whitespace/braces', 4,
+                      'when starting a new scope, { should be on a line by itself')
```

Comment:
Perhaps escaping out some single quotes to go around the brace, i.e. `'{'` just to make it obvious this is a character. Just sort of a visual thing and not necessary.